.github/workflows/updater.yml

```diff
@@ -1,35 +1,64 @@
 name: Check Virtualenv
 on:
-  # schedule:
-  #   - cron: "0 0 * * *"
-  workflow_dispatch:
+  schedule:
+    - cron: "0 0 * * *"
+  #workflow_dispatch:
 
 jobs:
-  check-for-updates:
+  publish:
+    name: Publish new tag
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        versionRegex:
+         - v2\.[0-9]+\.[0-9]+$
+
     steps:
     - uses: actions/checkout@v2.4.0
 
-    - name: Check for updates
-      id: update
-      run: |
-        version=$(curl --silent https://api.github.com/repos/actions/virtual-environments/releases | \
-          jq -r 'map(select(.tag_name|startswith("ubuntu20"))|.tag_name)[0]')
+    - uses: dysnix/find-latest-tag@v1
+      id: upstream
+      with:
+        regex: ${{ matrix.versionRegex }}
+        repository: actions/runner
 
-        echo ::set-output name=version::$version
-        if [[ "${version}a" = ubuntu20/*a ]];then
-          sed -i -r "s|^(ARG VIRTUAL_ENVIRONMENT_VERSION)\s*=.*|\1=$version|" Dockerfile
-        fi
+    - uses: dysnix/find-latest-tag@v1
+      id: latest
+      with:
+        compared-to-tag: ${{ steps.upstream.outputs.tag }}
+        regex: ${{ matrix.versionRegex }}
+        repository: ${{ github.repository }}
 
-    - name: Create PR
-      uses: peter-evans/create-pull-request@v3
+    - name: Publish new release
+      uses: ncipollo/release-action@v1
+      if: steps.latest.outputs.newer == 'true' || steps.latest.outputs.tag == ''
       with:
```

Comment:
@killmeplz So could you please clarify... We create a github release, i.e. release and tag and what happens after this?

Reply:
@killmeplz I mean how do we publish the docker image and run that corresponding workflow?

Reply:
'Build and Push' workflow is gonna run on tag creation

Reply:
Yes, it's what I would have thought too. However 'Build and Push' won't in github we can't use recursive invocation of workflows. In fact only this workflow will run and 'Build and Push' won't be(

Reply:
If you'd like to check this @killmeplz, plz let me know and I'll approve and merge this, but please move the workflow into a separate file first, so we do not override the original update from the upstream. We are not interested in solving merge conflicts in the future.